### PR TITLE
Add framework selection, show support by framework

### PR DESCRIPTION
**Pull request by kylesuss**
This PR adds:

* the framework selection menu
* messaging for code snippets when a snippet does not exist for a given framework
* styling for the unsupported feature banner

### CODE REVIEW

File: src/components/screens/DocsScreen/CodeSnippets.tsx [hunk 1]
```diff
@@ -61,18 +69,56 @@ export function PureCodeSnippets(props) {
...
+  withMissingMessaging,
+}) {
+  return (
+    <>
+      {withMissingMessaging && (
+        <MissingMessaging>
+          This snippet doesnt exist for {stylizeFramework(currentFramework)} yet.{' '}
+          <Link
+            appearance="secondary"
+            href="https://github.com/storybookjs/storybook/tree/next/docs"
```
Comment (tmeasday) [inline src/components/screens/DocsScreen/CodeSnippets.tsx:92]: Is it possible to get a link to the actual file here?

Also, I wonder if this should be `master` given that is the branch that is deployed?

Comment (tmeasday) [reply src/components/screens/DocsScreen/CodeSnippets.tsx:92]: Any thoughts on that @shilman? (i.e. do we want docs PRs to `next` or `master` for snippet contributions?)

Comment (kylesuss) [reply src/components/screens/DocsScreen/CodeSnippets.tsx:92]: The file doesn't exist yet, hence why we can't link to it. I asked about this in slack & whether or not we should just link to the react snippet, but it sounds like there is a plan to add a contributing guide for the snippets eventually. This seemed to be the interim solution. Let's see what @shilman has to say on the branch...

Comment (shilman) [reply src/components/screens/DocsScreen/CodeSnippets.tsx:92]: Ideally it should link to `next` and we'll patch it through to `master`.

We could also link to `master` which would be more intuitive for users, but I prefer forcing everything into `next` first for consistency with our code workflow.

We need to patch changes through one way or the other regardless of where we link, so neither direction saves work for us as maintainers AFAIK.

Comment (tmeasday) [reply src/components/screens/DocsScreen/CodeSnippets.tsx:92]: Ok fine by me

**Pull request merged**